Tasks/SonarQubePostTest/SonarQubeReportHandler.ps1

```diff
@@ -25,37 +27,113 @@ function UploadLegacySummaryMdReport
 	$sonarQubeOutDir = GetSonarQubeOutDirectory
 
 	# Upload the summary markdown file
-	$summaryMdPath = [System.IO.Path]::Combine($sonarQubeOutDir, "summary.md")
-	Write-Verbose "Looking for a summary report at $summaryMdPath"
+	$legacySummaryReportPath = [System.IO.Path]::Combine($sonarQubeOutDir, "summary.md")
+	Write-Verbose "Looking for a summary report at $legacySummaryReportPath"
 
-	if ([System.IO.File]::Exists($summaryMdPath))
+	if ([System.IO.File]::Exists($legacySummaryReportPath))
 	{
-		Write-Verbose "Uploading the summary.md file"
-        Write-Host "##vso[task.addattachment type=Distributedtask.Core.Summary;name=SonarQube Analysis Report;]$summaryMdPath"
+         UploadReport $legacySummaryReportPath
 	}
 	else
 	{
-		 Write-Warning "Could not find the summary report file $summaryMdPath"
+		 Write-Warning "Could not find the summary report file $legacySummaryReportPath"
 	}
 }
 
 function UploadSummaryReportInternal
 {
     Assert ((IsReportEnabled) -eq $true) "The summary report is disabled"
     Assert ((IsPrBuild) -eq $false) "Cannot produce a report because the analysis was done in issues mode"
-        
-    $qualityGateStatus = GetOrFetchQualityGateStatus
-        
-    #TODO: actually upload a report
-    Write-Host "The quality gate status is $qualityGateStatus"
+
+    Write-Host "Creating a summary report"        
+    $reportPath = CreateReport
+    Write-Host "Uploading the report"
+    UploadReport $reportPath
 }
 
 function IsReportEnabled
 {
-    $includeReport = GetTaskContextVariable "MSBuild.SonarQube.IncludeReport"
+    $includeReport = GetTaskContextVariable "MSBuild.SonarQube.Internal.IncludeFullReport"
```

Comment:
This is set by the user in the pre-build task and I get it here by reading a build variable.